### PR TITLE
ref #139: Use specific login form action (avoid second access denied …

### PR DESCRIPTION
**Pull request by UlrichKu**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | https://github.com/chameleon-system/chameleon-system/issues/139
| License       | MIT

### CODE REVIEW

File: Resources/views/snippets/common/userInput/form/formLoginStandard.html.twig [hunk 1]
```diff
@@ -22,7 +22,7 @@
     {% endif %}
     {% if not bIsLoggedIn %}
         <div class="loginForm">
-            <form accept-charset="utf-8" method="post" class="{% block formClass%}form-horizontal{% endblock%}">
+            <form accept-charset="utf-8" method="post" action="{{ loginFormAction is defined ? loginFormAction : '?' }}" class="{% block formClass%}form-horizontal{% endblock%}">
```
Comment (umulmrum) [inline Resources/views/snippets/common/userInput/form/formLoginStandard.html.twig]: I'd omit the whole `action` attribute if loginFormAction is not defined. Adding a question mark will lead to subtle routing problems (and isn't correct per se, as we like to call the unmodified URL on submit).

File: Resources/views/webModules/MTExtranet/accessDenied.view.php [hunk 2]
```diff
@@ -16,11 +17,23 @@
...
 }
 $oViewRenderer->AddSourceObject('sLoginSuccessURL', $sSuccessURL);
 
+/**
+ * @var $extranetConfig ExtranetConfigurationInterface
+ */
+$extranetConfig = ServiceLocator::get('chameleon_system_extranet.extranet_config');
+$loginUrl = $extranetConfig->getLink(ExtranetConfigurationInterface::PAGE_LOGIN);
+if (null !== $loginUrl && '' !== $loginUrl) {
+    $oViewRenderer->AddSourceObject('loginFormAction', $loginUrl);
```
Comment (umulmrum) [inline Resources/views/webModules/MTExtranet/accessDenied.view.php:34]: What is the intention behind explicitly setting the form action?

edit: Ah got it - OK. Do we also need a change in the success URL?

**Review by umulmrum — CHANGES_REQUESTED** [review]

File: Resources/views/snippets/common/userInput/form/formLoginStandard.html.twig (hunk 1 above)
Comment (UlrichKu) [reply Resources/views/snippets/common/userInput/form/formLoginStandard.html.twig]: I was wondering if that (calling original url) was really intended.

As far as I know not specifying action is considered an error in html.

File: Resources/views/webModules/MTExtranet/accessDenied.view.php (hunk 2 above)
Comment (UlrichKu) [reply Resources/views/webModules/MTExtranet/accessDenied.view.php:34]: Why in the success url?

Above that is the current (intially requested) url.

File: Resources/views/snippets/common/userInput/form/formLoginStandard.html.twig (hunk 1 above)
Comment (umulmrum) [reply Resources/views/snippets/common/userInput/form/formLoginStandard.html.twig]: The action attribute was required before HTML 5 only. Omitting would also make clear that it's really intended to call the current URL again.

File: Resources/views/webModules/MTExtranet/accessDenied.view.php (hunk 2 above)
Comment (umulmrum) [reply Resources/views/webModules/MTExtranet/accessDenied.view.php:34]: Nevermind, looks good.

**Comment by UlrichKu** [comment]
This might / is also relevant for all existing themes out there?

File: Resources/views/snippets/common/userInput/form/formLoginStandard.html.twig (hunk 1 above)
Comment (UlrichKu) [reply Resources/views/snippets/common/userInput/form/formLoginStandard.html.twig]: ok

**Review by umulmrum — APPROVED** [review]

**Comment by umulmrum** [comment]
Thanks @UlrichKu

**Pull request merged**